### PR TITLE
[sqllab] fix data grid's instant search function

### LINKED ISSUE #4713: [sql lab] search text box in query results isn't working

### PR DESCRIPTION
**Pull request by mistercrunch**
It looks like any non-string type would break the search feature.
of `FilterableTable`.
closes https://github.com/apache/incubator-superset/issues/4713

### CODE REVIEW

File: superset/assets/javascripts/modules/visUtils.js [hunk 1]
```diff
@@ -1,9 +1,13 @@
 export function getTextWidth(text, fontDetails = '12px Roboto') {
   const canvas = document.createElement('canvas');
   const context = canvas.getContext('2d');
-  context.font = fontDetails;
-  const metrics = context.measureText(text);
-  return metrics.width;
+  if (context) {
+    // Won't work outside of a browser context (ie unit tests)
+    context.font = fontDetails;
```
Comment (hughhhh) [inline superset/assets/javascripts/modules/visUtils.js]: nit: `context = {...context, font: fontDetails}`

File: superset/assets/javascripts/modules/visUtils.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+    const metrics = context.measureText(text);
+    return metrics.width;
```
Comment (hughhhh) [inline superset/assets/javascripts/modules/visUtils.js]: nit: just return `context.measureText(text).width`

File: superset/assets/javascripts/modules/visUtils.js (hunk 1 above)
Comment (hughhhh) [reply superset/assets/javascripts/modules/visUtils.js]: I tested it in my console:
```
> test_obj = { my_func: () => 'test'}
{ my_func: [Function: my_func] }

> new_test_obj = {...test_obj}
{ my_func: [Function: my_func] }

> new_test_obj
{ my_func: [Function: my_func] }

> new_test_obj.my_func()
'test'

```
It seems like it works, but no biggie if you want to keep it as is. I'm not 💯% sure it won't break at runtime.

Comment (mistercrunch) [reply superset/assets/javascripts/modules/visUtils.js]: I meant prototype funcs

**Pull request merged**